update-ubuntu-or-debian-server/run.sh

```
@@ -2,193 +2,973 @@
+fi
+
+# Function to prompt the user with enhanced visuals
 prompt_user() {
+    local question="$1"
+    local config_var="$2"
+    
     if [ "$unattended" = true ]; then
-        return 0  # Automatically proceed in unattended mode
+        # Check configuration for auto-answers
+        if [ -n "$config_var" ]; then
+            local config_value=$(eval echo \$${config_var})
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)